react/week5/my-app/src/components/GithubUserSearch.js

```diff
@@ -0,0 +1,134 @@
+import React, { useState, useEffect } from "react";
+import { Card, Icon, Image } from "semantic-ui-react";
+
+const GithubUserSearch = () => {
+  const [userInput, setUserInput] = useState("");
+  const [data, setData] = useState("");
+  const [loading, setLoading] = useState(false);
+  const [currentUserDetails, setCurrentUserDetails] = useState("");
+  const [error, setError] = useState(null);
+
+  const userClickHandler = (login) => {
+    fetch(`https://api.github.com/users/${login}`)
+      .then((res) => res.json())
+      .then((result) => {
+        console.log("RESULT", result);
+        const userDetails = (
+          <div>
+            <Card>
+              <Image src={result.avatar_url} wrapped ui={false} />
+              <Card.Content>
+                <Card.Header>{result.login}</Card.Header>
+              </Card.Content>
+              <Card.Content extra>
+                <a>
+                  <Icon name="user" />
+                  Bio: {result.bio}.
+                </a>
+              </Card.Content>
+              <Card.Content extra>
+                <a>
+                  <Icon name="user" />
+                  {result.public_repos} repos.
+                </a>
+              </Card.Content>
+              <Card.Content extra>
+                <a>
+                  <Icon name="user" />
+                  Location: {result.location}.
+                </a>
+              </Card.Content>
+            </Card>
+          </div>
+        );
+        setCurrentUserDetails(userDetails);
+      });
+  };
+
+  const searchHandler = (e) => {
+    setUserInput(e.target.value);
+  };
+
+  const searchSubmit = () => {
+    if (loading || userInput === "") {
+      return;
+    }
+    setLoading(true);
+    fetch(`https://api.github.com/search/users?q=${userInput}`)
+      .then((res) => res.json())
+      .then((data) => {
+        setLoading(false);
+        console.log("Data", data);
+        let dataStr = null;
+        if (data && data.message) {
+          dataStr = <li>{data.message}</li>;
+          setCurrentUserDetails("");
+        } else if (data && data.total_count > 0) {
+          dataStr = data.items.map((item) => {
+            return (
+              <li className="user_details" key={item.id}>
+                <a
+                  className="user_link"
+                  href="#{item.login}"
+                  onClick={(e) => userClickHandler(item.login)}
+                >
+                  {item.login}
+                </a>
+              </li>
```

Comment:
Here again you are building markup in the "login function". The markup should be in the render function